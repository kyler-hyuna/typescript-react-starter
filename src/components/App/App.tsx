import React from 'react'
const Image = require('../../assets/image.jpg')
import value from '../../assets/names.json'

import * as actions from '@actions'
import { Store } from '@types'
import { connect, Dispatch } from 'react-redux'

interface ConnectedProps {
  counter: number
}

interface ConnectedDispatch {
  increment: () => void
}

type Props = ConnectedProps & ConnectedDispatch

export class App extends React.Component<Props> {
  render() {
    return (
      <>
        <h1>{this.props.counter} new react projects</h1>

        {value.names.map((a: string) => <p key={a}>{a}</p>)}

        <img src={Image} alt="" />

        <button onClick={this.props.increment}>Hey</button>
      </>
    )
  }
}

const mapStateToProps = (state: Store) => ({
  counter: state.counter,
})

const mapDispatchToProps = (dispatch: Dispatch<Store>) => ({
  increment: () => dispatch(actions.increment()),
})

export default connect<ConnectedProps, ConnectedDispatch, {}, Props>(
  mapStateToProps,
  mapDispatchToProps
)(App)
