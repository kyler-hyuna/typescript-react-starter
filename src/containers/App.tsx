import React from 'react';
const Image = require('../assets/image.jpg');
import value from '../assets/names.json';

const App = () => (
  <>
    <h1>A new react project</h1>
    {value.names.map((a: string) => <p key={a}>{a}</p>)}
    <img src={Image} alt="" />
  </>
);

export default App;
