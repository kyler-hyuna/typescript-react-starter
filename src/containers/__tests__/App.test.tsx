import React from 'react';
import { create } from 'react-test-renderer';
import App from '../App';

it('it should work', () => {
  const tree = create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
