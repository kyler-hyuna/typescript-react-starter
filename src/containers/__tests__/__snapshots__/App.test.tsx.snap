// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should work 1`] = `
Array [
  <h1>
    A new react project
  </h1>,
  <p>
    na
  </p>,
  <p>
    ma
  </p>,
  <img
    alt=""
    src="test-file-stub"
  />,
]
`;
